### ISSUE TITLE
Pyright reports unsupported augmented assignment after resolving direct division to `Never`

### ISSUE THREAD

**Issue by cmp0xff** [microsoft/pyright#11648]
# Pyright reports unsupported augmented assignment after resolving direct division to `Never`

## Describe the bug

Pyright reports `reportOperatorIssue` for an augmented assignment expression using a subscript target when the left-hand side and right-hand side are both parameterized with `Any` and the relevant overload set contains a leading overload that returns `Never`.

In the example below, `data[0]` has type `Series[Any]` and the right-hand side has type `Array[Any, DType[Any]]`. The overload set for `Series.__truediv__` includes:

1. a broad array overload returning `Never`, representing an invalid array dtype case, followed by
2. valid numeric overloads returning `Series[float]`.

Pyright appears to select the leading `Never` overload for `data[0] / Array[Any, DType[Any]]()`. As a result, it rejects the augmented assignment `data[0] /= Array[Any, DType[Any]]()` as unsupported.

Other type checkers accept this example: `mypy`, `pyrefly`, and `ty` all report no errors.

## Expected behavior

Pyright should accept the augmented assignment, or at least should not resolve the operation through the `Never` overload when both the series scalar type and array scalar type are `Any` and later overloads are valid matches.

## Code or Screenshots

Minimal self-contained repro using one runtime module, one stub file, and one consumer file.

`mre_lib.py`

```python
class DType:
    pass


class Array:
    pass


class Series:
    pass


class Frame:
    pass
```

`mre_lib.pyi`

```python
from typing import Any, Generic, Never, TypeAlias, TypeVar, overload

Shape = TypeVar("Shape", covariant=True, default=Any)
DTypeT = TypeVar("DTypeT", bound="DType[Any]", covariant=True, default="DType[Any]")
ScalarT = TypeVar("ScalarT", default=Any)
ScalarT_co = TypeVar("ScalarT_co", covariant=True)

class DType(Generic[ScalarT_co]): ...

class Array(Generic[Shape, DTypeT]): ...

NDArray: TypeAlias = Array[Shape, DType[ScalarT]]
DateTimeArray: TypeAlias = NDArray[Shape, str]
IntArray: TypeAlias = NDArray[Shape, int]
FloatArray: TypeAlias = NDArray[Shape, float]

class Series(Generic[ScalarT]):
    @overload
    def __truediv__(self, other: DateTimeArray) -> Never: ...
    @overload
    def __truediv__(self: Series[int], other: IntArray) -> Series[float]: ...
    @overload
    def __truediv__(self: Series[float], other: FloatArray) -> Series[float]: ...

class Frame:
    def __getitem__(self, key: int) -> Series[Any]: ...
    def __setitem__(self, key: int, value: Series[Any]) -> None: ...
```

`ttest.py`

```python
from typing import Any

from mre_lib import Array, DType, Frame


data = Frame()

data[0] /= Array[Any, DType[Any]]()
```

Pyright output:

```sh
$ pyright ttest.py
.../ttest.py:8:1 - error: Operator "/=" not supported for types "Series[Any]" and "Array[Any, DType[Any]]" (reportOperatorIssue)
1 error, 0 warnings, 0 informations
```

For comparison:

```sh
$ mypy ttest.py
Success: no issues found in 1 source file

$ pyrefly check ttest.py
INFO 0 diagnostics

$ ty check ttest.py
All checks passed!
```

If `reveal_type(data[0] / Array[Any, DType[Any]]())` is added, Pyright reveals the direct division result as `Never`, which seems to explain why the augmented assignment is rejected.

## VS Code extension or command-line

Command-line Pyright:

```text
pyright 1.1.411
```

co-authored by copilot

**Comment by rchiodo** [comment]
This should be fixed by this PR https://github.com/microsoft/pyright/pull/11601. Please upgrade to 1.1.412

**Comment by cmp0xff** [comment]
Hi @rchiodo, thank you for the reply. I am not sure if here is the best place to ask, but apparently 1.1.412 is not on [pypi](https://pypi.org/project/pyright/) as of today.

<img width="2672" height="1432" alt="Image" src="https://github.com/user-attachments/assets/ff48b1f6-741e-4c21-86e8-b0b32d4f6745" />

**Comment by rchiodo** [comment]
We don't manage the PyPi version of Pyright. That's handled by a 3rd party. NPM however is updated by us. It's actually 1.1.413

https://www.npmjs.com/package/pyright
